05/pairs-with-given-sum.md

```diff
@@ -2,56 +2,164 @@
+
+<br>
+
+As you come up with observations and questions, assume that error handling for invalid data is managed outside the function. We want to focus on the core behavior of the function we will write. 
 
 ##### !end-question
 ##### !hint
 
-Consider the following for inspiration:
+Further questions to ask as you read through the problem statement and examples:
+- What is the goal of the function?
+- What are the types of the expected inputs and outputs?
+- Are there any restrictions on any of the inputs?
+  - For example: if any of the inputs are a list, do we know anything about how the list is ordered?
+- What do the examples show us about the data types and values that are allowed for our inputs?
+- What do the examples tell us about the return value in different scenarios?
+- Reflecting on the observations you have made so far, what questions would give you new information?
 
+<br>
+
+Consider the following for inspiration:
 - [About PSEs](../about-pses/about-pses.md)
 - [Our example PSE with example answers](../about-pses/example-pse.md)
-- Any past PSEs you may have
+- Previous PSEs
 
 ##### !end-hint
 ### !explanation
 
-Here are some example clarifying questions:
+One of many possible responses could look like:
 
-1. How does the function handle non-numeric elements in the list of `numbers`?
-1. What is the behavior for a non-numeric `target`?
-1. Can each of the numbers be used more than once in a pair that sums to the target?
-1. What is the behavior for an empty list?
+1. The problem statement and examples show that we should return the total number of pairs that create the "target", not the actual pairs themselves.
+    - This means that we only need to count matching pairs, we do not need to store or return them.
 
-##### !end-explanation
+2. Looking at the first example, given the inputs numbers = [1, 2, 4, 5] with target = 6, there are two unique pairs: [1, 5] and [2, 4] and the expected output is "2". 
+    - This means that however we iterate through the input "numbers" we should not double count pairs. I will need some way to keep track of the numbers that have been used in pairs as I go.
+
+3. Can a number can be reused in more than one pair?
+    - It isn't mentioned in the problem statement and the examples do not include repeated numbers or duplicate pairs. Since nothing says that numbers should be used more than once, I will assume a number can only be used in a single pair.
+    - If I am assuming a number can only be used in a single pair, as with the previous observation, I will need to keep track of the used numbers.
+
+4. The problem uses the phrase “list of whole numbers,” which usually means non-negative integers.
+    - If numbers are non-negative, I don’t need to worry about negative values in calculations or checking for negative complements
```

Comment:
If you choose to implement the suggestion I made about the integers then we'd probably need to remove this one or re-work it.